### PR TITLE
Create SDK and screens

### PR DESCRIPTION
**Pull request by sirichandanagangam**
## Jira ticket

https://oregonstate-innovationlab.atlassian.net/browse/LAB-13
https://oregonstate-innovationlab.atlassian.net/browse/LAB-12

## Proposed changes

* I have done SDK setup
* I have included couple of screens which I will be using in future for rearranging according to user input
<details><summary>Click here for Screenshots</summary>
<p>
<table>
  <tr>
    <td>
     </td>
<td></td>
<td></td>
<td></td>
</tr>

</p>
</details>

### CODE REVIEW

**Review by wbraynen — CHANGES_REQUESTED** [review]
Nice job putting up a PR!

Before I review your diff, here's what you'll want do first:

1. Have a PR description.
2. Have a good PR title. 
3. Clean up your commit history.
4. You want to configure your repo to require at least one approver on PRs.  Because otherwise, right now, you could just merge the PR even if it’s not approved.  Let me know if you are able to set this up on a free account—I don’t know if github charges for this feature.

### Re. (1):

Here is a template for ya:
```
## Jira ticket

<insert link>

## Proposed changes

* <insert a short description of the first thing you are proposing>
* <insert a short ... of the second thing ... >
```

### Re. (2):
A good title summarizes, in a few words, what this PR is all about.

### Re. (3):
If you plan to use the “Squash and merge” button to merge this PR, then you actually don’t need to do anything and this is totally fine as long as you have a good PR title.  That’s because your PR title, once you merge the commit, will then become your commit message.

### Re. (4):
You want to configure your repo to require at least one approver on PRs because otherwise, right now, you could just merge the PR even if it’s not approved.  Let me know if you are able to set this up on a free account—I don’t know if github charges for this feature.  If it costs money, then I guess we'll just live with what we have :).

Once you complete these items, re-request review here (there is a "re-request" circular icon next to my username in the approvers list) and slack me to let me know and I will review the code!

File: app/src/main/java/com/siri/RearrangeScreens/FourthActivity.kt
Comment (wbraynen) [inline app/src/main/java/com/siri/RearrangeScreens/FourthActivity.kt]: Why leave this dead code here?

File: RearrangeScreensModule/src/main/java/com/siri/RearrangeScreensSDK/ScreenUtils.kt
Comment (wbraynen) [inline RearrangeScreensModule/src/main/java/com/siri/RearrangeScreensSDK/ScreenUtils.kt]: Four nitpicks:

1. Dose the `stringNames` parameter accept only one name or many?  If just one screen name (and it's not an array here, so I'm guessing it's just one), then why plural?  Your method name and your parameter name is plural, but seems it's just to specify one screen name?

2. Why is the word `string` in the name of the variable?  We know it's a string from its type.  Why not simply `name` or `screenName`?

3. Why is `stringNames` null by default?  What would it mean to not set it to anything? 

4. Also, a nitpick, but... you might want to install a linter because your formatting seems inconsistent:

```
context : Context?
```
but
```
stringNames:String?
```
Which convention do you want to follow?  Spaces after/before a colon or no spaces at all?  Some options:
```
context:Context
```
or
```
context: Context
```
or 
```
context : Context
```

File: RearrangeScreensModule/src/test/java/com/siri/RearrangeScreens/ExampleUnitTest.kt
Comment (wbraynen) [inline RearrangeScreensModule/src/test/java/com/siri/RearrangeScreens/ExampleUnitTest.kt:16]: Ah, not a real test. You're keeping it as just an example?  (It's okay if you are; just want to understand:)

File: app/src/main/java/com/siri/RearrangeScreens/FourthActivity.kt
Comment (wbraynen) [inline app/src/main/java/com/siri/RearrangeScreens/FourthActivity.kt]: Install a linter.  You are using whitespace in funny ways :)

You don't have to install one in this PR.  You could instead create a ticket for your next sprint to install one.

File: app/src/main/java/com/siri/RearrangeScreens/ThirdActivity.kt
Comment (wbraynen) [inline app/src/main/java/com/siri/RearrangeScreens/ThirdActivity.kt:17]: Nitpick.

This will probably depend on your linter, but I personally find code easier to read when it's spaced out like this:

```
override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    setContentView(R.layout.activity_three)
    val button: Button = findViewById(R.id.simpleButton)

    button.setOnClickListener {
        val i = Intent(this, FourthActivity::class.java)
        startActivity(i)
    }
}
```
That way it's not just a wall of text and I can see clearly and quickly that (a) there is a call to your super class first and that (b) you are adding a closure as an event listener to your button.

**Review by wbraynen — COMMENTED** [review]
Left a few questions

File: app/src/main/java/com/siri/RearrangeScreens/MainActivity.kt
Comment (wbraynen) [inline app/src/main/java/com/siri/RearrangeScreens/MainActivity.kt:16]: I don't think `returnScreenOrder` helps you check `takeScreenNames` because your `returnScreenOrder` function returns hard-coded values instead of those that were passed into the `takeScreenNames` function:
```
    fun returnScreenOrder() {
         var screenOrderList: Array<String> = arrayOf("Screen3", "Screen1", "Screen2");
     }
```

So seems like either you meant to not hard-code the values in `returnScreenOrder` or you else why call `returnScreenOrder` here?

File: app/src/main/java/com/siri/RearrangeScreens/MainActivity.kt
Comment (wbraynen) [inline app/src/main/java/com/siri/RearrangeScreens/MainActivity.kt]: Any reason it's a variable (a `var`) and not a constant (a `val`)?

**Review by wbraynen — COMMENTED** [review]
Left a few comments.

File: RearrangeScreensModule/src/main/java/com/siri/RearrangeScreensSDK/ScreenUtils.kt
Comment (sirichandanagangam) [reply RearrangeScreensModule/src/main/java/com/siri/RearrangeScreensSDK/ScreenUtils.kt]: I addressed other parts of this comment,  coming to linter part, I will take it up next sprint

File: app/src/main/java/com/siri/RearrangeScreens/FourthActivity.kt
Comment (sirichandanagangam) [reply app/src/main/java/com/siri/RearrangeScreens/FourthActivity.kt]: Will install linter as part of a ticket in sprint as suggested

File: app/src/main/java/com/siri/RearrangeScreens/ThirdActivity.kt
Comment (sirichandanagangam) [reply app/src/main/java/com/siri/RearrangeScreens/ThirdActivity.kt:17]: I can format the same way  entire project once i use the linter

File: RearrangeScreensModule/src/test/java/com/siri/RearrangeScreens/ExampleUnitTest.kt
Comment (sirichandanagangam) [reply RearrangeScreensModule/src/test/java/com/siri/RearrangeScreens/ExampleUnitTest.kt:16]: That was just an example

**Review by wbraynen — APPROVED** [review]
Your first PR for this project behind you!  Good job!

**Pull request merged**
